cms/djangoapps/api/v1/serializers/discussion_settings.py

```diff
@@ -0,0 +1,72 @@
+""" Discussion settings serializers. """
+
+
+from datetime import datetime
+
+from django.utils.dateparse import parse_date, parse_datetime
+from rest_framework import serializers
+
+from models.settings.course_metadata import CourseMetadata
+from xmodule.modulestore.django import modulestore
+
+
+def to_datetime(val, timepart):
+    """
+    Converts a given string to a datetime object. If the string only
+    contains date part, than additional timepart param used to build
+    datetime object. Raises ValueError if unable to parse given string.
+    """
+
+    parsed_val = parse_datetime(val)
+    if not parsed_val:
+        parsed_val = parse_date(val)
+        if parsed_val:
+            parsed_val = datetime.combine(parsed_val, timepart)
+        else:
+            raise ValueError('Invalid date')
+    return parsed_val
+
+
+def blackout_date_range_validator(value):
+    """
+    Given two date/datetime string, checks if those are valid dates
+    and if 2nd date/datetime is larger than 1st date/datetime. Doesn't
+    perform any transformation though, returns ``value`` as is if valid,
+    else raises ``serializers.ValidationError``.
+    """
+
+    [from_date, to_date] = value
```

Comment:
If only one date is specified, this results in a value error: not enough values to unpack (expected 2, got 1)
Which isn't caught.

Reply:
Not applicable now as we are using existing validation.